### DISCUSSION TITLE
Keyboard Shortcut to Toggle Subtitle?

### DISCUSSION THREAD

**Discussion by Raza-0007** [smplayer-dev/smplayer#725] [General]
Is there a keyboard shortcut key to "toggle subtitle" in smplayer?

This is meant for video tracks that have multiple subtitles included, but not force-enabled by default.

Smplayer has a key for "next subtitle", but it only works if at least one subtitle was previously selected by the mouse. You can then use this key to go through the list of subtitles and even turn subtitles off, but you cannot use it to enable the subtitles if they were not forced enabled in the video track.

mpv itself has a key to toggle subtitles, from off state to on, and then it toggles through the list of subtitles and even turns subtitles off. It is very helpful in situations where you do not need to have subtitles on all the time, but want to enable them for some hard to understand dialogues. It is easier to toggle them on and off from keyboard rather than by a mouse.

So, is there such a key in smplayer? If not, then is there any chance of converting the "next subtitle" key into a "toggle subtitle" key that enables/toggles the subtitles from an off state?

Thanks,

**Comment by smplayer-dev** [comment]
You can assign any key to the "subtitle visibility" action in preferences -> keyboard and mouse.

**Reply by Raza-0007** [reply]
From my tests, the "subtitle visibility" key does not enable subtitles if they are off, it just makes the subtitles visible or invisible on the screen.

For example, I have a video in mkv with a subtitle track, but the track is off by default. In order for me to enable this subtitle track, I have to enable it by selecting it with a mouse. Both the "subtitle visibility" key and the "next subtitle" key do nothing. 

Now if the subtitle is enabled once by the mouse, then I can use these keys to toggle the subtitle on/off or to make it invisible.

In mpv, you can just use their keyboard shortcut "j by default" to toggle the subtitle on/cycle/off. You do not need a mouse. vlc player also has a similar key "v by default". smplayer unfortunately lacks this very important keyboard shortcut.

So the key I am asking for is something like a "cycle subtitle" or "toggle subtitle" key, where one of the options is to enable them from off to the first track, then next and next and then back to off. So "on/cycle/off" key.

Thanks,

**Comment by smplayer-dev** [comment]
Do you have a sample mkv file where this problem happens?

**Reply by Raza-0007** [reply]
The sample is attached below. The mkv file is in a zip file, as github does not support uploading mkv.

It is a 26 second clip from an anime. It has two subtitles tracks both labeled "Eng" or Track 1 and Track 2 based on the media player you play the file on. The first track has English subtitles and the second track is "sings and songs" only, so is blank in this clip. I just left it there to demonstrate the subtitle cycle process. Both tracks are off by default. User has to select them if they need them.

If you play this file in mpv, it plays without subtitles. But all you need to do is to hit the "j" key and the first track with English dialogue is enabled. Hit "j" again and the second track is enabled, and the subtitles turn off if you hit "j" the third time.

The same behavior occurs with vlc, but there the key is "v".

In smplayer however, the subtitles never come on by pressing the "next subtitle" key ("v" in my case) or "subtitle visibility" key ("k"in my case). The only way to enable the subtitle is to select one of them with a mouse. Once enabled this way, then you can use the above keys to toggle/cycle between the subtitles and even make them invisible. And since smplayer remembers the file settings by default, so next time when the file is played, the keys remain functional. If you delete the file's settings, smplayer reverts back to its earlier behavior, and the keys become useless. You again have to first select one of the subtitles with a mouse to enable the keys.

I was wondering if a key shortcut like the mpv's or vlc's can be implemented in smplayer so one does not have to use the mouse to enable the subtitles.


Thanks,

 

[Sample Video.zip](https://github.com/smplayer-dev/smplayer/files/11710780/Sample.Video.zip)

**Comment by smplayer-dev** [comment]
Thanks, I'll investigate the bug.
I noticed that pressing Shift+K (which is the shortcut for the previous subtitle) the subtitles eventually appear.

**Reply by Raza-0007** [reply]
You are right about the "previous subtitle" key. Its works as expected. Even though it enters the subtitle list from the bottom and works its way back to the top. Slightly counter intuitive but it works. It never even occurred to me to check it. This would mean that there is some sort of a glitch in the "next subtitle" key. 

I am going to use the "previous subtitle" key while you are fixing the "next subtitle" key.

Thanks for your help.

**Comment by smplayer-dev** [comment]
I think the problem is fixed in this commit https://github.com/smplayer-dev/smplayer/commit/23a177c6802850b1776594d33571a79add595cb4

**Reply by Raza-0007** [reply]
Thanks for the quick fix. I will wait for the next Windows build to test it, as I do not know how to compile smplayer myself. 

I am currently using the "smplayer-portable-22.7.0.10131-x64-qt5.6.7z" development build that you complied 2 weeks ago with mpv 0.35.1 and it has been working great. I have not noticed any issues. 

Thanks for all the hard work to provide us with an excellent front end for the mpv player.

~Raza

**Comment by smplayer-dev** [comment]
I've just built the package [smplayer-portable-22.7.0.10143-x64-qt5.6.7z](https://github.com/smplayer-dev/smplayer/suites/13518531936/artifacts/743147744)

**Reply by Raza-0007** [reply]
Thanks,

I tested it and the "next subtitle" key works perfectly!

~Raza
